enterprise/frontend/src/embedding-sdk/components/public/MetabotQuestion/MetabotChatEmbedding.tsx

```diff
@@ -0,0 +1,193 @@
+import cx from "classnames";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { t } from "ttag";
+
+import { useSdkDispatch } from "embedding-sdk/store";
+import {
+  Box,
+  Flex,
+  Icon,
+  Loader,
+  Textarea,
+  Tooltip,
+  UnstyledButton,
+} from "metabase/ui";
+import { MetabotIcon } from "metabase-enterprise/metabot/components/MetabotIcon";
+import { useMetabotAgent } from "metabase-enterprise/metabot/hooks";
+import { resetConversationId } from "metabase-enterprise/metabot/state";
+
+import Styles from "./MetabotChatEmbedding.module.css";
+
+const MIN_INPUT_HEIGHT = 42;
+
+interface MetabotChatEmbeddingProps {
+  onResult: (result: Record<string, any> | null) => void;
+}
+
+const EMBEDDING_METABOT_ID = "c61bf5f5-1025-47b6-9298-bf1827105bb6";
+
+export const MetabotChatEmbedding = ({
+  onResult,
+}: MetabotChatEmbeddingProps) => {
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  const [input, setMessage] = useState("");
+
+  const metabot = useMetabotAgent();
+
+  const resetInput = useCallback(() => {
+    setMessage("");
+    setInputExpanded(false);
+  }, []);
+
+  const metabotRequestPromiseRef = useRef<{ abort: () => void } | null>(null);
+
+  const handleSend = () => {
+    const trimmedInput = input.trim();
+    if (!trimmedInput.length || metabot.isDoingScience) {
+      return;
+    }
+    const metabotRequestPromise = metabot.submitInput(
+      trimmedInput,
+      EMBEDDING_METABOT_ID,
+    );
+    metabotRequestPromiseRef.current = metabotRequestPromise;
+
+    metabotRequestPromise
+      .then(onResult)
+      .catch((err) => console.error(err))
```

Comment:
I suppose we can add error handling in the future, ok for now

Reply:
I didn't touch this part by the way. But yeah, we could do more here.